src/main.rs

```diff
@@ -9,6 +9,7 @@ mod index_manager;
 mod models;
 mod post;
 mod put;
+mod sparse;
```

Comment:
Nitpick (and I don't know if it's consistently applied in this codebase), but I'd rather have the cfg attribute here than in the `sparse.rs` file.

This way I know directly from the main file whether the module is relevant or not

```suggestion
#[cfg(feature = "sparse-index")]
mod sparse;
```